microlith/loki/alerting/couchbase/couchbase-rules.yaml

```
@@ -76,6 +76,23 @@ groups:
           description: Node {{ $labels.hostname }} has reported errors due to long connection names (known issue MB-34280).
           remediation: Reduce the lengths of your nodes' hostnames, or contact Couchbase Technical Support.
 
+
+      - alert: CB90042-segmentationFaults
+        expr: sum( (count_over_time({file="diag.log"} |= "segfault") + (count_over_time({file="couchbase.log"} |= "segfault") )>0
```

I think you need a time for `count_over_time`:
```suggestion
        expr: sum( (count_over_time({file="diag.log"} |= "segfault") + (count_over_time({file="couchbase.log"} |= "segfault" [1h]) )>0
```